src/pages/level/Level.tsx

```diff
@@ -19,15 +24,59 @@ const Level: FC<LevelProps> = () => {
   })
 
   const settings = useSettings()
+  const blocklyHandle = useRef<BlocklyWorkspaceHandle | null>(null)
 
   return useParamsRequired({
     shape: { id: yup.number().required().min(1) },
     children: () => (
       <Box sx={{ display: "flex" }}>
         <Controls />
         <Box component="main" sx={{ flexGrow: 1 }}>
-          <Typography>Settings: {JSON.stringify(settings)}</Typography>
-          <Typography>Level state: {JSON.stringify(level)}</Typography>
+          {/* TODO: fix style */}
+          <PanelGroup
+            direction="horizontal"
+            style={{ height: "100vh", width: "100%" }}
+            onLayout={() => {
+              // Trigger Blockly workspace resize after layout change
+              if (blocklyHandle.current) {
+                blocklyHandle.current.resize()
+              }
+            }}
+          >
+            <Panel
+              id="blockly-panel"
+              style={{ width: "100%" }}
+              defaultSize={50}
+              minSize={20}
+            >
+              <BlocklyWorkspace
+                ref={blocklyHandle}
+                toolboxContents={[
+                  { kind: "block", type: "logic_compare" },
+                  { kind: "block", type: "logic_compare" },
+                  { kind: "block", type: "logic_compare" },
+                ]}
```

Comment:
please add this to the level state. this will be retrieved from the backend per level. side note: the backend returns keys using snake_case (since it's Python).
```suggestion
                toolboxContents={level.toolbox_contents}
```